model_finetuning/PushIOProcessor.py

```diff
@@ -0,0 +1,265 @@
+import html
+from datetime import datetime, timedelta
+
+import requests
+from playhouse.shortcuts import model_to_dict, dict_to_model
+from requests.adapters import HTTPAdapter, Response
+from requests.packages.urllib3.util.retry import Retry
+from configparser import ConfigParser
+
+from db import *
+
+
+class PushIOProcessor:
+	"""
+	A class for downloading from PUSHIO and writing directly to a DB as efficiently as possible without using a
+	out of process mechanism.
+	"""
+
+	def __init__(self, start_date, end_date, subreddit, db_instance, min_score=5):
+		self.score = min_score
+		self.start_date_time = datetime.fromisoformat(start_date)
+		self.end_date_time = datetime.fromisoformat(end_date)
+		self.subreddit = subreddit
+		self.session = requests.Session()
+		self.db = db_instance
+
+	@staticmethod
+	def set_is_url_only(comment: dict) -> bool:
+		comment_body = comment.get('body')
+		return (comment_body.startswith('[') and comment_body.endswith(')')) or (
+				'http' in comment_body.lower() and ' ' not in comment_body)
+
+	@staticmethod
+	def get_data_from_response(response: Response) -> [dict]:
+
+		if response.status_code != 200:
+			logging.error("Response has non 200 status code")
+			return []
+
+		try:
+			json_content = response.json()
+			data = json_content.get('data')
+			if data:
+				return data
+			else:
+				logging.error("No Data in Response")
+				return []
+		except Exception as e:
+			logging.error(e)
+			return []
+
+	@staticmethod
+	def clean_text(text) -> str:
+		if text is None:
+			text = ""
+		# have to unescape it twice, for reason I don't fully understand
+		text = html.unescape(text)
+		text = html.unescape(text)
+		# Strip and whitespace off of the end
+		text = text.strip()
+		return text
+
+	def loop_between_dates(self):
```

Comment:
It would be better to pass `start_date` and `end_date` into this function, as a best practice. It keeps it more versatile and you can do unit testing on the function in isolation, if you want to in the future.

Reply:
I actually find it simpler, in this case, to call the class with the parameters to unit test the function (but as I have not done  that). I can make it static. 

The reasoning here and throughout was to encapsulate the search parameters as members of the class. It's how I would design something like this in a strictly typed OOP language. 

I am moving to a static invocation.